src/printer.js

```diff
@@ -2843,7 +2843,11 @@ function printBinaryishExpressions(path, parts, print, options, isNested) {
     // the other ones since we don't call the normal print on BinaryExpression,
     // only for the left and right parts
     if (isNested && node.comments) {
-      parts.push(comments.printComments(path, p => "", options));
+      parts.splice(
+        0,
+        parts.length,
+        comments.printComments(path, p => concat(parts.slice()), options)
+      );
```

Comment:
https://github.com/prettier/prettier/blob/master/src/printer.js#L223

The outer call site creates the part array, passes it as reference and looks at it at the end to do the indentation and to print it.